src/qamqpclient.cpp

```diff
@@ -660,14 +660,20 @@ QAmqpExchange *QAmqpClient::createExchange(int channelNumber)
 QAmqpExchange *QAmqpClient::createExchange(const QString &name, int channelNumber)
 {
     Q_D(QAmqpClient);
+    if (!name.isEmpty() && d->exchanges.contains(name)
+                    && (!d->exchanges[name].isNull()))
```

Comment:
Yep, I figured that's what was meant.  I read though it's not wise to subclass the container classes since they don't have virtual destructors.

http://wiki.qt.io/Coding_Conventions#Things_to_avoid

That suggests keeping the container safely stashed and exposing a subset of the methods.  In fact, really all we need is the operator[] and contains() methods, since we're not likely to iterate over the list.